insights/v1alpha1/tests/datagathers.insights.openshift.io/InsightsOnDemandDataGather.yaml

```
@@ -14,6 +14,104 @@ tests:
+            persistentVolume:
+              mountPath: /data
+              persistentVolumeClaim:
+                name: test-claim
+    - name: Should be able to create a DataGather with a storageSpec Epemeral
+      initial: |
+        apiVersion: insights.openshift.io/v1alpha1
+        kind: DataGather
+        spec:
+          storageSpec:
+            type: Ephemeral
+      expected: |
+        apiVersion: insights.openshift.io/v1alpha1
+        kind: DataGather
+        spec:
+          storageSpec:
+            type: Ephemeral
+    - name: When storageSpec.type is PersistentVolumeClaim then PersistentVolume must be present
+      initial: |
+        apiVersion: insights.openshift.io/v1alpha1
+        kind: DataGather
+        spec:
+          storageSpec:
+            type: PersistentVolumeClaim
+      expectedError: 'spec.storageSpec: Invalid value: "object": persistentVolume is required when type is PersistentVolumeClaim, and forbidden otherwise'
+    - name: When storageSpec.type is not PersistentVolumeClaim then PersistentVolume must not be present
+      initial: |
+        apiVersion: insights.openshift.io/v1alpha1
+        kind: DataGather
+        spec:
+          storageSpec:
+            type: Ephemeral
+            persistentVolume:
+              persistentVolumeClaim:
+                name: test-claim
+      expectedError: 'spec.storageSpec: Invalid value: "object": persistentVolume is required when type is PersistentVolumeClaim, and forbidden otherwise'
+    - name: mountPath must not contain colon
+      initial: |
+        apiVersion: insights.openshift.io/v1alpha1
+        kind: DataGather
+        spec:
+          storageSpec:
+            type: PersistentVolumeClaim
+            persistentVolume:
+              mountPath: /data:/data
+              persistentVolumeClaim:
+                name: test-claim
+      expectedError: 'spec.storageSpec.persistentVolume.mountPath: Invalid value: "string": mountPath must not contain a colon'
+    - name: storageSpec.type is required
+      initial: |
+        apiVersion: insights.openshift.io/v1alpha1
+        kind: DataGather
+        spec:
+          storageSpec:
+            persistentVolume:
+              mountPath: /data
+              persistentVolumeClaim:
+                name: test-claim
+      expectedError: "spec.storageSpec.type: Required value"
+    - name: storageSpec.persistentVolume.persistentVolumeClaim can not be empty
+      initial: |
+        apiVersion: insights.openshift.io/v1alpha1
+        kind: DataGather
+        spec:
+          storageSpec:
+            type: PersistentVolumeClaim
+            persistentVolume:
+              mountPath: /data
+              persistentVolumeClaim:
+      expectedError: 'spec.storageSpec.persistentVolume.persistentVolumeClaim: Required value, <nil>: Invalid value: "null": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation'
```

The required value is the important part that you are checking with this test right?
```suggestion
      expectedError: 'spec.storageSpec.persistentVolume.persistentVolumeClaim: Required value'
```